src/Modules/Guilds/Hooks/useVotingPowerForProposalExecution.ts

```diff
@@ -0,0 +1,93 @@
+import { useMemo } from 'react';
+import { BigNumber } from 'ethers';
+import { useContractRead } from 'wagmi';
+import useGuildImplementationType from 'Modules/Guilds/Hooks/useGuildImplementationType';
+import { WagmiUseContractReadResponse } from 'Modules/Guilds/Hooks/types';
+import useSnapshotId from 'Modules/Guilds/Hooks/useSnapshotId';
+import { useTypedParams } from 'Modules/Guilds/Hooks/useTypedParams';
+import ERC20GuildContract from 'contracts/ERC20Guild.json';
+import SnapshotERC20Guild from 'contracts/SnapshotERC20Guild.json';
+import SnapshotRepERC20Guild from 'contracts/SnapshotRepERC20Guild.json';
+
+/**
+ * -  ERC20GuildContract
+ *    function getVotingPowerForProposalExecution() public view virtual returns (uint256) {}
+ *
+ * -  SnapshotERC20Guild:
+ *    function getVotingPowerForProposalExecution(uint256 snapshotId) public view virtual returns (uint256) {}
+ *
+ * -  SnapshotRepERC20Guild
+ *    function getSnapshotVotingPowerForProposalExecution(bytes32 proposalId) public view virtual returns (uint256) {}
+ */
+
+interface UseVotingPowerForProposalExecutionProps {
+  contractAddress: string;
+  proposalId?: string;
+}
+
+const getConfig = ({
+  isRepGuild,
+  isSnapshotGuild,
+  contractAddress,
+  snapshotId,
+  _proposalId,
+}) => {
+  const baseErc20Config = {
+    enabled: !!contractAddress,
+    addressOrName: contractAddress,
+    contractInterface: ERC20GuildContract.abi,
+    functionName: 'getVotingPowerForProposalExecution',
+  };
+
+  const snapshotConfig = {
+    enabled: !!contractAddress && !!snapshotId,
+    addressOrName: contractAddress,
+    contractInterface: SnapshotERC20Guild.abi,
+    functionName: 'getVotingPowerForProposalExecution(uint256)',
+    args: [snapshotId?.toString()],
+  };
+
+  const snapshotRepConf = {
+    enabled: !!contractAddress && !!_proposalId,
+    addressOrName: contractAddress,
+    contractInterface: SnapshotRepERC20Guild.abi,
+    functionName: 'getSnapshotVotingPowerForProposalExecution(bytes32)',
+    args: [_proposalId],
+  };
+  // Validate args to avoid null values
+  if (isRepGuild && isSnapshotGuild && !!_proposalId) return snapshotRepConf;

```

Comment:
💡 I wonder if we could add this to the other hooks as well to avoid having null

Reply:
Not following. What are "this" and "other hooks"?